src/routes/safe/store/actions/transactions/__tests__/utils/transactionHelpers.test.ts

```diff
@@ -0,0 +1,875 @@
+    const transaction = getMockedTxServiceModel({ to: safeAddress2, value: '0' })
+
+    // when
+    const result = isInnerTransaction(transaction, safeAddress)
+
+    // then
+    expect(result).toBe(false)
+  })
+  it('It should return true if the transaction recipient is the given safeAddress and the txValue is 0', () => {
+    // given
+    const transaction = makeTransaction({ recipient: safeAddress, value: '0' })
+
+    // when
+    const result = isInnerTransaction(transaction, safeAddress)
+
+    // then
+    expect(result).toBe(true)
+  })
+  it('It should return false if the transaction recipient is the given safeAddress and the txValue is >0', () => {
+    // given
+    const transaction = makeTransaction({ recipient: safeAddress, value: '100' })
+
+    // when
+    const result = isInnerTransaction(transaction, safeAddress)
+
+    // then
+    expect(result).toBe(false)
+  })
+  it('It should return false if the transaction recipient is not the given safeAddress', () => {
+    // given
+    const transaction = makeTransaction({ recipient: safeAddress2, value: '100' })
+
+    // when
+    const result = isInnerTransaction(transaction, safeAddress)
+
+    // then
+    expect(result).toBe(false)
+  })
+})
+
+describe('isCancelTransaction', () => {
+  const safeAddress = '0xdfA693da0D16F5E7E78FdCBeDe8FC6eBEa44f1Cf'
+  it('It should return false if given a inner transaction with empty data', () => {
+    // given
+    const transaction = getMockedTxServiceModel({ to: safeAddress, value: '0', data: null })
+
+    // when
+    const result = isCancelTransaction(transaction, safeAddress)
+
+    // then
+    expect(result).toBe(true)
+  })
+  it('It should return false if given a inner transaction without empty data', () => {
+    // given
+    const transaction = getMockedTxServiceModel({ to: safeAddress, value: '0', data: 'test' })
+
+    // when
+    const result = isCancelTransaction(transaction, safeAddress)
+
+    // then
+    expect(result).toBe(false)
+  })
+})
+
+describe('isPendingTransaction', () => {
+  it('It should return true if the transaction is on pending status', () => {
+    // given
+    const transaction = makeTransaction({ status: TransactionStatus.PENDING })
+    const cancelTx = makeTransaction({ data: null })
+
+    // when
+    const result = isPendingTransaction(transaction, cancelTx)
+
+    // then
+    expect(result).toBe(true)
+  })
+  it('It should return true If the transaction is not on pending status but his cancellation transaction yes', () => {
+    // given
+    const transaction = makeTransaction({ status: TransactionStatus.AWAITING_CONFIRMATIONS })
+    const cancelTx = makeTransaction({ status: TransactionStatus.PENDING })
+
+    // when
+    const result = isPendingTransaction(transaction, cancelTx)
+
+    // then
+    expect(result).toBe(true)
+  })
+  it('It should return true If the transaction is not pending status and his cancellation transaction also is not on that status', () => {
```

Comment:
It should return true If the transaction and a cancellation transaction are not pending